lib/src/dap_resolver.dart

```diff
@@ -9,11 +13,152 @@ class DapResolver {
   DapResolver(this.didResolver, {http.Client? client})
       : client = client ?? http.Client();
 
-  Future<void> getMoneyAddresses(Dap dap) async {}
+  Future<List<MoneyAddress>> getMoneyAddresses(Dap dap) async {
+    throw UnimplementedError();
+  }
+
+  Future<DapResolutionResult> resolve(Dap dap) async {
+    final registryDidResolution = await didResolver.resolveDid(dap.registryDid);
+    if (registryDidResolution.hasError()) {
+      throw DapResolutionException(
+          'Failed to resolve registry DID: ${registryDidResolution.didResolutionMetadata.error}');
+    }
+
+    final registryServices = registryDidResolution.didDocument!.service ?? [];
+    final dapRegistryService = registryServices.firstWhere(
+      (service) => service.type == 'DAPRegistry',
+      orElse: () => throw DapResolutionException(
+          'Registry DID does not have a DAPRegistry service'),
+    );
+
+    if (dapRegistryService.serviceEndpoint.isEmpty) {
+      throw DapResolutionException(
+          'DAPRegistry service does not have a service endpoint');
+    }
```

Comment:
nit:
You could grab the endpoint here so you don't need to use `first` below if you want.

```dart
final serviceEndpoint = dapRegistryService.serviceEndpoint.firstOrNull;
if (serviceEndpoint == null) {
  throw DapResolutionException('DAPRegistry service does not have a service endpoint');
}

// then below
registryEndpoint = Uri.parse(serviceEndpoint)
```